src/memory/memory.ts

```diff
@@ -59,4 +56,15 @@ export class Memory {
   getSegmentSize(segment: number): number {
     return this.values[segment]?.length ?? 0;
   }
+
+  printMemory() {
```

Comment:
isn't it more idiomatic to call this `toString()`, return a `string` and eventually use it like

```
console.log(memory.toString())
```

?

Reply:
Indeed, much better this way
Made something similar for printing `relocatedMemory`